src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.cs

```diff
@@ -553,12 +564,48 @@ internal static void DecrementRunningForeground(Thread thread)
             if (!thread.IsBackground) 
             {
                 DecrementRunningForeground();
+                Debug.Assert(_foregroundRunningCount >= 1, "_foregroundRunningCount is less than 1");
             }
         }
 
         internal static void WaitForForegroundThreads() 
-        {
-            _allDone.WaitOne();
+        {         
+            Thread current = Thread.CurrentThread;
```

Comment:
Can this be just:
```
```
Thread.CurrentThread.IsBackground = true;

if (_foregroundRunningCount == 0)
   return;

... the rest ...
```
?